map/mapObject.mjs

```diff
@@ -0,0 +1,102 @@
+"use strict"
+export {MapObject, MapIcon}
+
+import {worldSpaceToMapSpace, mapSpaceToScreenSpace, iconH} from "../map.js"
+import {Point} from "./point.mjs"
+
+/**
+ * An object that will be displayed on the map canvas.
+ */
+function MapObject(){
+    this.minX = 0;
+    this.maxX = 0;
+    this.minY = 0;
+    this.maxY = 0;
+}
+/**
+ * Does this object contain the specified point?
+ * @param {Point} point point to check
+ */
+MapObject.prototype.contains = function(point){
+    if(point == null){
+        return false;
+    }
+    return (this.minX < point.x && point.x < this.maxX && this.minY < point.y && point.y < this.maxY);
+}
+MapObject.prototype.width = function(){
+    return this.maxX - this.minX;
+}
+MapObject.prototype.height = function(){
+    return this.maxY - this.minY;
+}
+
+/**
+ * Construct a map location object from a location json cell.
+ * @param {*} cell 
+ */
+function MapIcon(cell){
+    MapObject.call(this);
+
+    this.cell = cell;
+    this.recalculateBoundingBox();
+    if(cell.hive.classname == "nirnroot"){
+        this.icon = iconSwitch("Nirnroot");
+    }
+    else{
+        this.icon = iconSwitch(cell.icon);
+    }
+}
+MapIcon.prototype = Object.create(MapObject.prototype);
+
+/**
+ * whenever we zoom, we will need to call this.
+ */
+MapIcon.prototype.recalculateBoundingBox = function(){
+    let mapCoords = worldSpaceToMapSpace(new Point(this.cell.x, this.cell.y));
```

Comment:
zooming in/out changes the map coordinates , so we need to recalculate them.

Reply:
Why do we store these in map coords if they need to be recalculated on zoom, AND the display coordinates need to be recalculated on draw?
The conversion from world space to map space is more expensive than the conversion from map space to screen space. And this does change less often than screen space coords, so it's a midpoint between the two.